solr/core/src/java/org/apache/solr/response/DocsStreamer.java

```diff
@@ -142,6 +142,11 @@ public boolean hasNext() {
     return docIterator.hasNext();
   }
 
+  // called at the end of the stream
```

Comment:
the reason for `prepare` and `finish` was to have the same naming of components, imo it makes easier to understand what's going on, I do like the Closeable idea, do you think it would be better?

> Also, line 89 still calls setContext() .. is it right? or I'm missing something?

No, I missed line 89, thank you